robot_modeling/DQ_SerialManipulatorDH.m

```diff
@@ -202,13 +160,152 @@
                 (d*cosine_of_half_alpha*cosine_of_half_theta)/2.0...
                 - (a*sine_of_half_alpha*sine_of_half_theta  )/2.0
                 ]);
+         end        
+        
+    end
+    
+    methods
+        function obj = DQ_SerialManipulatorDH(A,convention)
+            % These are initialized in the constructor of
+            % DQ_SerialManipulator
+            %obj.convention = convention;
+            %obj.n_links = size(A,2);
+              
+            obj = obj@DQ_SerialManipulator(size(A,2));
+            obj.dh_matrix_ = A;
+            %obj.theta = A(1,:); %obj.dh_matrix_(1,:);
+            %obj.d = A(2,:);     %obj.dh_matrix_(2,:);
+            %obj.a = A(3,:);     %obj.dh_matrix_(3,:);
+            %obj.alpha = A(4,:); %obj.dh_matrix_(4,:);
+            
+            if nargin == 0
+                error('Input: matrix whose columns contain the DH parameters')
+            end            
+            if nargin == 2
+                warning('DQ_SerialManipulatorDH(A,convention) is deprecated. Please use DQ_SerialManipulatorDH(A) instead.');
+                
+            end
+            
+            if(size(A,1) ~= 5)
+                error('Input: Invalid DH matrix. It should have 5 rows.')
+            end
+            
+            % Add type
+            %obj.type = A(5,:);
+        end      
+                      
+       
+        
+        function th = get_thetas(obj)
+            %GET_THETAS() returns the first row of the Matrix dh_matrix_, which
+            % correspond to the parameter 'theta' in the DH convention.
+            th = obj.dh_matrix_(1,:); %obj.theta;
         end
         
-        function w = get_w(obj,ith)       
-            if obj.type(ith) == obj.JOINT_ROTATIONAL
-                w = DQ.k;
+        function ds = get_ds(obj)
+            % GET_DS() returns the second row of the Matrix dh_matrix_, which
+            % correspond to the parameter 'd' in the DH convention.
+            ds = obj.dh_matrix_(2,:); %obj.d;
+        end
+        
+        function as = get_as(obj)
+            % GET_AS() returns the third row of the Matrix dh_matrix_, which
+            % correspond to the parameter 'a' in the DH convention.
+            as = obj.dh_matrix_(3,:); %obj.a;
+        end
+        
+        function alphas = get_alphas(obj)
+            % GET_ALPHAS() returns the fourth row of the Matrix dh_matrix_, which
+            % correspond to the parameter 'alpha' in the DH convention.
+            alphas =  obj.dh_matrix_(4,:); %obj.alpha;            
+        end
+        
+        function types = get_types(obj)
+            % GET_TYPES() returns the fifth row of the Matrix dh_matrix_, which
+            % correspond to the type of joints of the robot: 
```

Comment:
which corresponds

Reply:
According to some requested modifications, now we have:
`% GET_JOINT_TYPES() Returns the joint type, which can be either REVOLUTE or PRISMATIC.`
This is temporal since the name of the joints (I mean, JOINT_REVOLUTE, REVOLUTE_JOINT, REVOLUTE, etc) has not been chosen yet.

Reply:
Since MATLAB has an [enumeration class](https://uk.mathworks.com/help/matlab/enumeration-classes.html), it suffices to create a class called JointType, and then we can have just the types REVOLUTE and PRISMATIC.